### PR TITLE
Fix #8239: BlockUI: DRY JavaScript code

### LINKED ISSUE #8239: BlockUI: DRY JavaScript code
Duplicate code can be moved to function and `PrimeFaces.ajax.Utils.getSourceId` can be used.

### PR DESCRIPTION
**Pull request by jepsar**
Fix #8239

### CODE REVIEW

**Review by melloware — APPROVED** [review]

File: primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js [hunk 1]
```diff
@@ -53,60 +53,61 @@ PrimeFaces.widget.BlockUI = PrimeFaces.widget.BaseWidget.extend({
...
+     */
+    isXhrSourceATrigger: function (settings) {
+        var sourceId = PrimeFaces.ajax.Utils.getSourceId(settings);
+        if (!sourceId) {
+            return false;
+        }
+        // we must evaluate it each time as the DOM might has been changed
+        var triggers = PrimeFaces.expressions.SearchExpressionFacade.resolveComponents(this.cfg.triggers);
+
+        return $.inArray(sourceId, triggers) !== -1 && !this.cfg.blocked;
```
Comment (tandraschko) [inline primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js]: please fast escape in the method when !this.cfg.blocked

Comment (tandraschko) [reply primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js]: Netter remove the cfg.blocked check completely from this Methode and check outside

File: primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js [hunk 2]
```diff
@@ -53,60 +53,66 @@ PrimeFaces.widget.BlockUI = PrimeFaces.widget.BaseWidget.extend({
...
+
+    /**
+     * Checks whether one of component's triggers equals the source ID from the provided settings.
+     *
+     * @param {JQuery.AjaxSettings} settings containing source ID.
+     * @returns {boolean} `true` if the widget is not configured to be blocked ans if one of component's triggers equals
+     * the source ID from the provided settings.
+     */
+    isXhrSourceATrigger: function (settings) {
+        if (this.cfg.blocked) {
```
Comment (tandraschko) [inline primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js]: thats wait i meant, that i would move this check outside the method as "isXhrSourceATrigger" is independent of "cfg.blocked"

Comment (jepsar) [reply primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js]: Alright.. was doubting what you meant with your last comment. It just struck me that "netter" was probably German (it's the same in Dutch) 😃

Comment (tandraschko) [reply primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js]: :D it should be ´better´, i shouldnt use my smartphone sometimes...

**Comment by tandraschko** [comment]
i assume its well tested?

**Comment by jepsar** [comment]
Tested with the showcase using Chrome, Firefox and Safari on macOS.

**Pull request merged**